mathesar/models/base.py

```diff
@@ -283,6 +283,60 @@ class Explorations(BaseModel):
     description = models.CharField(null=True)
 
 
+class Form(BaseModel):
+    id = models.UUIDField(primary_key=True)
+    name = models.CharField()
+    version = models.IntegerField()
+    database = models.ForeignKey('Database', on_delete=models.CASCADE)
+    base_table_oid = models.PositiveBigIntegerField()
+    schema_oid = models.PositiveBigIntegerField()
+    description = models.CharField(null=True)
+    header = models.OneToOneField('FormHeader', on_delete=models.DO_NOTHING)
+    submission = models.ForeignKey('SubmissionSettings', on_delete=models.CASCADE)
+    is_public = models.BooleanField(default=False)
+
+
+class FormHeader(BaseModel):
+    title = models.CharField()
+    subtitle = models.CharField(null=True)
+
+
+class SubmissionSettings(BaseModel):
+    role = models.ForeignKey('ConfiguredRole', on_delete=models.CASCADE)
+    message = models.CharField(null=True)
+    redirect_url = models.URLField(null=True)
+    submit_label = models.CharField(null=True)
+
+
+class FormField(BaseModel):
+    id = models.CharField(primary_key=True)
+    attnum = models.SmallIntegerField()
+    table_oid = models.PositiveBigIntegerField()
+    related_form = models.ForeignKey('Form', on_delete=models.CASCADE, related_name='form_field')
+    kind = models.CharField(
+        choices=[
+            ("scalar_column", "scalar_column"),
+            ("foreign_key", "foreign_key"),
+            ("reverse_foreign_key", "reverse_foreign_key")
+        ],
+    )
+    record_summary_template = models.JSONField(null=True)
+    label = models.CharField(null=True)
+    help = models.CharField(null=True)
+    placeholder = models.CharField(null=True)
+    needs_validation = models.BooleanField(default=False)
```

Comment:
This is not accurate. This should be either contain a list of settings for what kind of validations to perform, or we should just have an `is_required` field here. 

The `BasicValidation` interface contains an `is_required` field which is a type of validation which specifies if we require the end-user to enter a value. This is a form level validation performed in the frontend/service layers.

Reply:
> The validation should only be performed in one layer or the other, no? Which should it be?

If there's a NOT NULL constraint on the underlying column, that'll error out at the db layer. This field has nothing to do with that.

This is for columns that allow NULL on the db, but the user wants to ensure that the end-user specifies data for it while filling the form.

This check happens on the frontend for providing immediate feedback, and also the service layer as part of validating incoming requests.